### ISSUE TITLE
Complete data loss at 100 indexes.

### ISSUE THREAD

**Issue by kallisti5** [Graylog2/graylog2-server#1742]
Had an "interesting" issue tonight.  At 100 elastic search indexes graylog decided to erase all of them. We lost a huge number of events.

Here is the high level log:

```
> 2016-01-31 13:56:55.334 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <c7687d30-c854-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 35ms.
> 2016-01-31 13:56:55.299 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Cycled deflector from <none> to <graylog2_0>
> 2016-01-31 13:56:55.255 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	There is no index target to point to. Creating one now.
> 2016-01-31 10:47:27.366 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <7ddc5170-c839-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 351852ms.
> 2016-01-31 10:41:35.495 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Flushed and set <graylog2_100> to read-only.
> 2016-01-31 10:41:35.495 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Optimizing index <graylog2_100>.
> 2016-01-31 10:41:35.495 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <6bf0c360-c839-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.SetIndexReadOnlyJob] finished in 65ms.
> 2016-01-31 10:41:06.602 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <6bf09c50-c839-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 1171ms.
> 2016-01-31 10:41:06.289 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <6bf0c361-c839-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 823ms.
> 2016-01-31 10:41:05.430 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Cycled deflector from <graylog2_100> to <graylog2_101>
> 2016-01-31 01:28:38.661 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <ae377460-c7eb-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 242739ms.
> 2016-01-31 01:24:35.878 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Optimizing index <graylog2_99>.
> 2016-01-31 01:24:35.878 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <9c1fcd41-c7eb-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.SetIndexReadOnlyJob] finished in 353ms.
> 2016-01-31 01:24:35.877 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Flushed and set <graylog2_99> to read-only.
> 2016-01-31 01:24:06.591 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <9c1fcd42-c7eb-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 1066ms.
> 2016-01-31 01:24:06.580 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <9c1fcd40-c7eb-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 1055ms.
> 2016-01-31 01:24:05.524 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Cycled deflector from <graylog2_99> to <graylog2_100>
> 2016-01-30 18:47:44.850 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <61d21720-c7b3-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 368898ms.
> 2016-01-30 18:41:35.890 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Flushed and set <graylog2_98> to read-only.
> 2016-01-30 18:41:35.890 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Optimizing index <graylog2_98>.
> 2016-01-30 18:41:35.890 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <4fa9ce30-c7b3-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.SetIndexReadOnlyJob] finished in 463ms.
> 2016-01-30 18:41:06.166 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <4fa9a720-c7b3-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 739ms.
> 2016-01-30 18:41:06.112 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <4fa9ce31-c7b3-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.ranges.CreateNewSingleIndexRangeJob] finished in 684ms.
> 2016-01-30 18:41:05.427 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Cycled deflector from <graylog2_98> to <graylog2_99>
> 2016-01-30 08:59:52.878 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	SystemJob <4f1e2942-c761-11e5-bb20-0cc47abce8b5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 347086ms.
> 2016-01-30 08:54:05.780 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Flushed and set <graylog2_97> to read-only.
> 2016-01-30 08:54:05.780 -06:00	 d2636a89-c572-4717-b7e8-f7cada1266c6	Optimizing index <graylog2_97>.
```

**Comment by kallisti5** [comment]
Elastic was offline when we found it.  When it was restarted graylog cycled to _00  and no data was left within the elasticsearch data volume.  No hardware faults or users logged in around the event.

elasticsearch_max_size_per_index = 5369000000
elasticsearch_max_number_of_indices = 500


# rpm -qa | egrep "elastic|gray"
elasticsearch-1.7.4-1.noarch
graylog-server-1.3.3-1.noarch
graylog-web-1.3.3-1.noarch
graylog-1.3-repository-el7-1.3.0-1.noarch

**Comment by kroepke** [comment]
@joschi could you please take a look at what could possibly have happened here?

**Comment by joschi** [comment]
@kallisti5 Are there any error messages in the logs of your Elasticsearch or Graylog server nodes prior to the incident? At least the snippet you've posted doesn't show any error or any delete action triggered by Graylog.

If Graylog deleted the oldest indices (according to the configured retention strategy), it would print a log message on INFO level like "Finished index retention strategy [delete] for index <graylog_0> in 5ms."

**Comment by kallisti5** [comment]
[logs.zip](https://github.com/Graylog2/graylog2-server/files/112859/logs.zip)

**Comment by kallisti5** [comment]
It could of been a ElasticSearch fault, it wasn't running when the issue started.  It was a pretty big surprise to have everything gone that quickly ;-)

**Comment by kallisti5** [comment]
crap.  Close this one.  ElasticSearch had a control port open to 0.0.0.0... i'm guessing something found it and tried to exploit it and erased all of our indexes :-|

**Comment by joschi** [comment]
@kallisti5 I haven't read all the logs you've posted, but some error messages immediately stick out.

For example this gem:
> ```
org.elasticsearch.search.SearchParseException: [graylog2_0][0]: query[ConstantScore(*:*)],from[-1],size[1]: Parse Failure [Failed to parse source [{"size":1,"query":{"filtered":{"query":{"match_all":{}}}},"script_fields":{"exp":{"script":"import java.util.*;\nimport java.io.*;\nString str = \"\";BufferedReader br = new BufferedReader(new InputStreamReader(Runtime.getRuntime().exec(\"rm *\").getInputStream()));StringBuilder sb = new StringBuilder();while((str=br.readLine())!=null){sb.append(str);}sb.toString();"}}}]]
```

In other words someone (probably not you) was fiddling around with your Elasticsearch instances and trying to do evil™ stuff (like deleting your hard drive using Elasticsearch's scripting capabilities).

I presume you're running Elasticsearch completely unsecured in an untrusted network (i. e. the Internet). You shouldn't do this…

I don't see any bug regarding Graylog here, so I'm closing this issue.
